core/src/avm1/activation.rs

```diff
@@ -1868,7 +1868,9 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     fn action_set_target(&mut self, target: &str) -> Result<FrameControl<'gc>, Error<'gc>> {
         let base_clip = self.base_clip();
         let new_target_clip;
-        let root = base_clip.root();
+        let root = base_clip
+            .root()
+            .expect("AVM1 display objects must have root");
```

Comment:
Let's avoid these `except`s on `base_clip.root()` -- although this should never happen in AVM1, I have gotten paranoid about panics when we could instead fail more gracefully. Let's do `.ok_or(...)?`, which will still output a message if there's an error.

Perhaps this could even be a util function or extension trait, because it is used so often: `base_clip.avm1_root()?`.

Reply:
I added `avm1_root` and a new `Error` type for this particular error condition.